gstreamer-rtsp-server/src/auto/r_t_s_p_session_pool.rs

```diff
@@ -0,0 +1,152 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir @ 9ae0a4a+)
+// from gir-files (https://github.com/gtk-rs/gir-files @ ???)
+// DO NOT EDIT
+
+use RTSPSession;
+use ffi;
+use glib;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::boxed::Box as Box_;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct RTSPSessionPool(Object<ffi::GstRTSPSessionPool, ffi::GstRTSPSessionPoolClass>);
+
+    match fn {
+        get_type => || ffi::gst_rtsp_session_pool_get_type(),
+    }
+}
+
+impl RTSPSessionPool {
+    pub fn new() -> RTSPSessionPool {
+        assert_initialized_main_thread!();
+        unsafe {
+            from_glib_full(ffi::gst_rtsp_session_pool_new())
+        }
+    }
+}
+
+impl Default for RTSPSessionPool {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+unsafe impl Send for RTSPSessionPool {}
+unsafe impl Sync for RTSPSessionPool {}
+
+pub trait RTSPSessionPoolExt {
+    fn cleanup(&self) -> u32;
+
+    fn create(&self) -> Option<RTSPSession>;
+
+    fn create_watch(&self) -> Option<glib::Source>;
```

Comment:
Needs manual implementation (like all functions returning a GSource)

Reply:
Done, needs review for sure